src/VoteProxy2.sol

```diff
@@ -0,0 +1,48 @@
+pragma solidity ^0.4.24;
+
+import 'ds-token/token.sol';
+import 'ds-chief/chief.sol';
+
+contract VoteProxy2 {
+    address public cold;
+    address public hot;
+    DSToken public GOV;
+    DSToken public IOU;
```

Comment:
IOU is never used except to `IOU.approve`.  I'm assuming the approval is necessary for some reason, but there is no need for the storage variable as you can just inline it all in the constructor: `chief.IOU().approve(chief, uint256(-1));`

Reply:
Useful for a UI to show the address of the IOU token.

Reply:
@MicahZoltu That's a good point, makes me thing of a partially related concern - is it even theoretically possible to exhaust this allowance? Even if you had a lot of MKR and you really tried? I originally had `reApprove` but removed it after some back of envelope math - someone sanity check

Reply:
Some tokens special case `uint256(-1)` as "infinite", I'm not sure about these two.  However, `unint256(-1)` cannot be exhausted in a human lifetime last I checked for a token that has 10^18 decimals.

Reply:
@MicahZoltu ds-token has infinite allowance with `uint256(-1)`

https://github.com/dapphub/ds-token/blob/master/src/token.sol#L49

Reply:
thank you